### PR TITLE
week5: presentation proposal

### PR DESCRIPTION
**Pull request by hampfh**
# Assignment Proposal

## Title

Comparing the differential mechanisms of declarative Terraform and imperative AWS infrastructure as code.

## Names and KTH ID

- Hampus Hallkvist (hallkvi@kth.se)
- Martin Lindefors (lindefor@kth.se)

## Deadline

- Week 5

## Category

- Presentation

## Description

In this presentation we aim to dive deep into the differences between declaring infrastructure as imperative and declarative, discuss the benefits and drawbacks of the very common tool terraform and compare it to AWS SDK. During the presentation we will have two examples set up side by side.

**Relevance**

While it is widely accepted that infrastructure as code is necessary to some extent there are many approaches. Better understanding the pros/cons of the structurally different approaches is necessary to make an informed decision.

### CODE REVIEW

**Comment by Deee92** [comment]
Sounds good, merging.

**Comment by Deee92** [comment]
Hi @hampfh, just noticed that you say presentation but you've added your proposal to `week5/demo/`. Moving to `week5/presentation/`, please push to the correct folder for future tasks.

**Comment by hampfh** [comment]
Thanks for notifying, I'll change it to presentation!

**Comment by Deee92** [comment]
I'd changed it myself. Good to remember for next time though. :)

**Comment by einbergisak** [comment]
# Feedback from Isak Einberg (ieinberg@kth.se)
I certify that generative AI, incl. ChatGPT, has not been used to write this feedback. Using generative AI without permission is considered academic misconduct.

I want to thank Hampus and Martin for letting me give them feedback for their presentation. I won’t be able to attend their presentation, but I hope that the feedback that I give on their presentation material (slides and script) that was provided to me, will be beneficial for them.

## High-level positives
- **The content is very clear.** I especially like that you use short bullet points throughout the entire presentation, instead of including longer texts or explanations.
- **The presentation is very visually appealing**. It's great that you have included lots of graphics and screenshots.
- **Your topic is very relevant for Week 5**. It is great that it explains modern and popular tools, with a relevant comparison.

## High-level negatives
- **Small image texts.** Your screenshots are very appreciated, but it would be beneficial to leave out some code in the images in favor of larger text.
- **Vague take-home message**. Your take-home message is very general, and does not specifically relate to your presentation topic.
- **Inconsistent styling**.  Font size and text spacing could be adjusted for better readability. It is currently inconsistent across the presentation.

## Slide Placement & Text Readability

The introductory slide is currently placed before the table of contents. I think it would be beneficial to include it after you have outlined the content of the presentation, in order to provide better context. On this introductory slide, titled “Infrastructure is hard…”, you have included a nice graphic that gets the attention of the audience. However, I do think that your bullet points are too small. It might look better on a big projector, but I would still recommend you to increase the text size to make it more readable. I think the text size on your other slides is adequate. 

On your “AWS SDK” slide, the readability of the bullet points is poor due to the congested items. It would be beneficial to add some spacing between each list item, in order to improve readability. But in general, I commend you that you stick to using bullet points on your slides instead of longer texts! Your slides are definitely not too wordy, and I think this design choice will definitely help get your point across, without the audience having to focus on reading instead of listening to what you are saying. This is consistent throughout all slides - great job!

## Font Style & Consistency

Another thing regarding the text, is that I noticed that your choice of font differs between different slides. This does not impact the understandability of your presentation, but it would be nice with a more coherent styling. I understand that this might be a result of parallel work. But I also noticed that you do a great job of matching the text color to the background! I was amazed by the “Declarative Terraform” slide, where the two words of the heading are in different colors as it spans two different background colors. I didn’t notice it at first, which is why I think that it was a very great choice in regards to clarity. Well done!

## Code Examples
It’s great that you showcase code examples in your presentation. However, the screenshots are not very readable. I would highly suggest that you limit your screenshots to a smaller part of the code, so that you can display it in a greater size. Otherwise, it will be very difficult for the audience to decode what is written. And for the “SQS Queue”, “EC2 + SG” and “SSH Connection” slides, the heading font size could be shrinked in favor of larger pictures. A suggestion would otherwise be to embed code into your slides. Here’s a resource for that if you are curious: https://www.thebricks.com/resources/how-to-embed-code-in-google-slides. 

I also really appreciate how you have blurred out your secrets/keys in your screenshots - you can definitely never be too careful, and it also serves as a great reminder to the audience! I think it’s good that you go back-and-forth in the script regarding who’s presenting. 

## Take Home Message
I would suggest you take a second look at the take home message. I think the current one is too general and unspecific. My suggestion is to include something for the audience to take home in regards to infrastructure as code specifically, to really tie in to the content of your presentation! 

## Background
You mention “EC2 instances” in multiple parts of your script/slides. It could be beneficial to include a brief introduction to what EC2 is, as well as what AWS in general is, as the audience might not be very familiar with the platforms. A resource for this could be Amazon’s own introductory pages (https://aws.amazon.com/ec2/ec2-get-started/, https://aws.amazon.com/ec2/features//).

Other than that, I think you do a good job of explaining what infrastructure of code is, and your introductory slide does a great job at explaining the background as to why it is relevant!
## Overall Presentation Quality

As a whole, I really like your presentation. The amount of pictures and graphics, combined with the use of bullet points instead of long texts, will definitely be beneficial in getting - and maintaining - the attention of the audience. Your slide template is really sleek and modern, and I think the tech youngsters in the audience will definitely appreciate it!

I also think it’s great that you go back-and-forth in the script regarding who’s presenting. It will definitely contribute to a great presentation flow!

## Summary
To briefly summarize: I think the introductory slide would be better placed after the table of contents, with larger bullet points for readability. Consistent fonts and improved spacing between items on the "AWS SDK" slide would also help. Code screenshots could be clearer, and the take-home message should focus more on infrastructure as code. 

Overall, the design, visuals, and teamwork seems great, with strong attention to detail - like blurring secrets in screenshots!

As a whole, I really like your presentation. I am confident that you guys will do a great job!

...and hopefully this feedback can help you do an even better job ;)

Good luck!
